Client/Frontend/Browser/BrowserViewController/BrowserViewController+WKNavigationDelegate.swift

```diff
@@ -99,6 +99,43 @@ extension BrowserViewController: WKNavigationDelegate {
             decisionHandler(.cancel)
             return
         }
+        
+        let isSafeBrowsingEnabled = { () -> Bool in
+            let isPrivateBrowsing = PrivateBrowsingManager.shared.isPrivateBrowsing
+            let domain = Domain.getOrCreate(forUrl: url, persistent: !isPrivateBrowsing)
+            return domain.isShieldExpected(.SafeBrowsing)
+        }
+        
+        if isSafeBrowsingEnabled() {
+            var safeBrowsingResult: SafeBrowsingResult = .safe
+            let semaphore = DispatchSemaphore(value: 0)
+            SafeBrowsingClient.shared.find(url.hashPrefixes()) { result, error in
+                defer { semaphore.signal() }
+                
+                if let error = error {
+                    log.error(error)
+                    safeBrowsingResult = result
+                    return
+                }
+                
+                safeBrowsingResult = result
+            }
+            
+            _ = semaphore.wait(timeout: .now() + .seconds(30))
+            
+            // Three types of results.. "safe", "dangerous", "unknown"
+            // In order not to accidentally block websites due to an error, if it's status is unknown, we should possibly show a different web-page that lets them know we couldn't determine if the page was safe or not..
+            //
+            // For now, we only block for sure when it's dangerous..
+            // According to the spec, unknown is supposed to be treated as safe, no matter what.
+            // However, browsers like Microsoft Edge will let the user know it cannot determine if a URL is safe.
```

Comment:
I believe that Edge is using Microsoft's version of Safe Browsing and not Google's so the ToS don't apply to them.